src/System.Memory/src/System/Buffers/ReadOnlyBuffer.cs

```diff
@@ -0,0 +1,377 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#pragma warning disable 1591
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+namespace System.Buffers
+{
+    /// <summary>
+    /// Represents a buffer that can read a sequential series of bytes.
+    /// </summary>
+    public readonly partial struct ReadOnlyBuffer<T>
+    {
+        private const int IndexBitMask = 0x7FFFFFFF;
+
+        internal const int MemoryListStartMask = 0;
+        internal const int MemoryListEndMask = 0;
+
+        internal const int ArrayStartMask = 0;
+        internal const int ArrayEndMask = 1 << 31;
+
+        internal const int OwnedMemoryStartMask = 1 << 31;
+        internal const int OwnedMemoryEndMask = 0;
+
+        internal readonly SequencePosition BufferStart;
+        internal readonly SequencePosition BufferEnd;
+
+        public static readonly ReadOnlyBuffer<T> Empty = new ReadOnlyBuffer<T>(new T[0]);
```

Comment:
Can we use the cached value `SpanHelpers.PerTypeValues<T>.EmptyArray` and `Array.Empty<T>()` for .NET Core?

Like this:
https://github.com/dotnet/corefx/blob/9340e12eb4cc5936edf0752f0771f05ccb6749c8/src/Common/src/CoreLib/System/MemoryDebugView.cs#L44-L48

Reply:
Not sure why I get 
```
System\Buffers\ReadOnlyBuffer.cs(35,92): error CS0117: 'SpanHelpers' does not contain a definition for 'PerTypeValues' [D:\github\Corefx\src\System.Memory\src\System.Memory.cspr oj]
```

Reply:
Are you targeting netcoreapp? Did you put it under #if?

SpanHelpers.PerTypeValues<T> is only available for portable span. For netcoreapp,  use Array.Empty<T>().